daml-lf/interface/src/main/scala/com/digitalasset/daml/lf/iface/DefDataType.scala

```diff
@@ -36,20 +40,23 @@ object DefDataType {
     }
 }
 
-sealed trait DataType[+RF, +VF] extends Product with Serializable {
-  def bimap[C, D](f: RF => C, g: VF => D): DataType[C, D] =
+sealed trait DataType[+RT, +VT] extends Product with Serializable {
+  def bimap[C, D](f: RT => C, g: VT => D): DataType[C, D] =
     Bifunctor[DataType].bimap(this)(f, g)
 
-  def fold[Z](record: Record[RF] => Z, variant: Variant[VF] => Z): Z = this match {
+  def fold[Z](record: Record[RT] => Z, variant: Variant[VT] => Z): Z = this match {
     case r @ Record(_) => record(r)
     case v @ Variant(_) => variant(v)
   }
 }
 
 object DataType {
 
-  /** Alias for application to [[FieldWithType]]. */
-  type FWT = DataType[FieldWithType, FieldWithType]
+  /** Alias for application to [[Type]]. Note that FWT stands for "Field with
+    * type", because before we parametrized over both the field and the type,
+    * while now we only parametrize over the type.
+    */
```

Comment:
The original reason for letting you parameterize the strings too was so you could have a separate type for pre-name-mangled field names and post-name-mangled field names (i.e. Scala identifiers); if you don’t think that is a sufficiently interesting use case then this is fine.

Reply:
it's not that i do not find the case sufficiently interesting, is that this abstraction does not work anymore where there are datatypes where you have _only_ labels (enums).

i think if we want to accomodate such definitions this change is the best compromise.

Reply:
I see; as long as we're clear that "not every label has an associated type" then that seems fine.

We could also introduce a third type parameter for enums...it's traditional to evolve to the final form of `Trifunctor` right? ;)